src/thingset_bin.c

```diff
@@ -141,6 +161,44 @@ static int bin_serialize_path(struct thingset_context *ts,
                : -THINGSET_ERR_RESPONSE_TOO_LARGE;
 }
 
+static int bin_serialize_metadata(struct thingset_context *ts,
+                                  const struct thingset_data_object *object)
+{
+    int err = bin_serialize_map_start(ts);
+    if (err) {
+        return err;
+    }
+
+    const char *name = "name";
+    if (!zcbor_tstr_encode_ptr(ts->encoder, name, 4)) {
+        return -THINGSET_ERR_RESPONSE_TOO_LARGE;
+    }
+
+    if (!zcbor_tstr_encode_ptr(ts->encoder, object->name, strlen(object->name))) {
```

Comment:
Would this also work? Maybe the strlen can be evaluated more efficiently if done during serialization.
```suggestion
    if (!zcbor_tstr_put_term(ts->encoder, object->name)) {
```